dhcpv4/client.go

```diff
@@ -27,10 +27,23 @@ var (
 	DefaultWriteTimeout = 3 * time.Second
 )
 
+// ClientExchangeOptions specifies a set of options to be used for a particular
+// exchange.
+type ClientExchangeOptions struct {
```

Comment:
How about removing this and using unicast when the RemoteAddr is != nil?

Reply:
Mainly to allow people to actually play with the option - nothing stops you from using a broadcast socket and specify a remote address. It *may* break. Probably we should warn out when we possibly have broken options combined though.

Reply:
I would keep it simple and infer if we need to use broadcast or unicast based on the RemoteAddr. Let's see what @insomniacslk thinks about it.

Reply:
I initially suggested to @gvialetto to leave the freedom to specify what kind of socket they want, since this library is inspired by scapy's principle of letting the user fiddle with the protocol (yet having sane defaults).
On the other side I appreciate simplicity. We could start simple - and on a second thought this is probably a good idea.

However checking for `RemoteAddr != nil` is not enough: to be done properly we have to check if the destination is a broadcast address, and to do that we need the subnet mask

Reply:
Another option is moving this bool to the Client struct. We already setup some configurations options in there such as read and write timeout.